contracts/SponsorFaucet.sol

```diff
@@ -23,96 +23,176 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
     uint256 public collateral_bound;
     //upper bound for single tx
     uint256 public upper_bound;
-    
-    mapping(address=>detail) public dapps;
-    
-    event applied(address indexed applicant, address indexed dapp, uint256 indexed amount);
-    
-    SponsorWhitelistControl internal_sponsor = SponsorWhitelistControl(0x0888000000000000000000000000000000000001);
-    
-    constructor(uint256 gasTotalLimit, uint256 collateralTotalLimit, uint256 gasBound, uint256 collateralBound, uint256 upperBound) public {
-        require(upperBound.mul(1000) <= gasBound, 'upperBound too high');
+
+    mapping(address => detail) public dapps;
+
+    event applied(
+        address indexed applicant,
+        address indexed dapp,
+        uint256 indexed amount
+    );
+
+    SponsorWhitelistControl internal_sponsor = SponsorWhitelistControl(
+        0x0888000000000000000000000000000000000001
+    );
+
+    constructor(
+        uint256 gasTotalLimit,
+        uint256 collateralTotalLimit,
+        uint256 gasBound,
+        uint256 collateralBound,
+        uint256 upperBound
+    ) public {
+        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
```

Comment:
Better to define as constant variable, or add necessary comment for this magic number.

Reply:
comment added